### PR TITLE
Add Flow

### PR DESCRIPTION
**Pull request by dashed**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,6 +3,8 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "flow-annotation-check": "^1.7.0",
```
Comment (dat2) [inline package.json]: Could you move these packages to devDependencies:
```
flow-annotation-check
flow-bin
husky
lint-staged
```

File: package.json [hunk 2]
```diff
@@ -28,7 +30,9 @@
     "test": "react-scripts test --env=jsdom",
     "eject": "react-scripts eject",
     "pretty": "prettier --single-quote --write 'src/**/*.js'",
-    "precommit": "lint-staged"
+    "precommit": "lint-staged",
+    "flow": "flow",
+    "flow-annotation-check": "cd src/;flow-annotation-check --list-files=noflow"
```
Comment (dat2) [inline package.json:32]: Would `flow-annotation-check --list-files=noflow src` work?

File: src/Root/component.js [hunk 3]
```diff
@@ -10,7 +12,14 @@ import NotFound from '../NotFound';
 import Dashboard from '../Dashboard';
 import Accounts from '../Accounts';
 
-class Root extends Component {
+type Props = {
```
Comment (dat2) [inline src/Root/component.js]: Should this be RootProps?

File: package.json (hunk 2 above)
Comment (dashed) [reply package.json:32]: nah I had tried that initiall, but for some reason I get this error with that command. but i worked around it by just doing `cd src/;`

```
$ npm run flow-annotation-check

> hledger-web-react@0.1.0 flow-annotation-check /Users/me/aaa/projects/js/hledger-web-react
> flow-annotation-check --list-files=noflow src

Report error: { error:
   { Error: Command failed: flow ast Accounts/actions.js
File not found: /Users/me/aaa/projects/js/hledger-web-react/Accounts/actions.js

    at ChildProcess.exithandler (child_process.js:272:12)
    at ChildProcess.emit (events.js:159:13)
    at maybeClose (internal/child_process.js:943:16)
    at Socket.stream.socket.on (internal/child_process.js:363:11)
    at Socket.emit (events.js:159:13)
    at Pipe._handle.close [as _onclose] (net.js:558:12)
     killed: false,
     code: 10,
     signal: null,
     cmd: 'flow ast Accounts/actions.js' },
  stdout: '',
  stderr: 'File not found: /Users/me/aaa/projects/js/hledger-web-react/Accounts/actions.js\n' }
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! hledger-web-react@0.1.0 flow-annotation-check: `flow-annotation-check --list-files=noflow src`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the hledger-web-react@0.1.0 flow-annotation-check script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/me/.npm/_logs/2017-12-10T20_39_53_480Z-debug.log
```

Comment (dat2) [reply package.json:32]: Ah, thats weird!

**Pull request merged**